code/game/objects/items/weapons/material/material_armor.dm

```diff
@@ -249,6 +323,47 @@ Protectiveness | Armor %
 	else
 		..()
 
+//Make plating inserts for modular armour.
+/obj/item/weapon/material/armor_plating/insert/attackby(var/obj/O, mob/user)
+	if(istype(O, /obj/item/weapon/weldingtool))
+		var /obj/item/weapon/weldingtool/S = O
+		if(S.remove_fuel(0,user))
+			if(!src || !S.isOn()) return
+			to_chat(user, "<span class='notice'>You trim down the edges to size.</span>")
+			user.drop_from_inventory(src)
+			var/obj/item/clothing/accessory/material/makeshift/light/new_armor = new(null, src.material.name)
+			user.put_in_hands(new_armor)
+			qdel(src)
```

Comment:
Probably should return at the end of each of these blocks to avoid weird behavior down the call chain.

Reply:
Done, thank you!